airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -45,10 +52,10 @@ def package_name_from_class(cls: object) -> str:
         raise ValueError(f"Could not find package name for class {cls}")
 
 
-class IncrementalMixin(ABC):
-    """Mixin to make stream incremental.
+class StateMixin(ABC):  # Rename to CheckpointMixin?
```

Comment:
I renamed this mixin because state setter/getter are no longer specific to Incremental streams. But to retain backwards compatibility I left the old `IncrementalMixin` but it just inherits from this mixin